tests/integration/conftest.py

```diff
@@ -287,3 +288,31 @@ def authenticated_platform_admin_client(
     login_user(user)
 
     yield anonymous_client
+
+
+@contextmanager
+def _count_sqlalchemy_queries() -> Generator[list[QueryInfo], None, None]:
+    queries: list[QueryInfo] = []
+    num_existing_queries = len(get_recorded_queries())
+
+    yield queries
+
+    new_queries = get_recorded_queries()
+    queries.extend(new_queries[num_existing_queries:])
+
+
+@pytest.fixture
+def track_sql_queries() -> t.Callable[[], _GeneratorContextManager[list[QueryInfo], None, None]]:
+    """This is a slight hack as we rely on `get_recorded_queries` from flask-sqlalchemy, but we use
+    `flask-sqlalchemy-lite` in the app code. However, we still have `flask-sqlalchemy` as a transitive dependency
+    and its `get_recorded_queries` still works for our purposes.
+
+    This is "ok" but not ideal - eventually we'll want to remove `flask-sqlalchemy` from our dependencies altogether,
+    and then this might stop working. We might try and get `get_recorded_queries` added to flask-sqlalchemy-lite:
+    https://github.com/pallets-eco/flask-sqlalchemy-lite/pull/21
```

Comment:
Is all of this addressed by lifting and shifting that into this repository/ extension?